aea/decision_maker/base.py

```diff
@@ -752,50 +752,43 @@ def _handle_tx_message_for_signing(self, tx_message: TransactionMessage) -> None
         :param tx_message: the transaction message
         :return: None
         """
-        if "contract" in tx_message.tx_id:
-            if self._is_acceptable_for_deployment(tx_message):
-                tx_signature = self._sign_contract_tx(tx_message=tx_message)
+        if self._is_acceptable_for_signing(tx_message):
+            if self._is_valid_message(tx_message):
+                tx_signature = self._sign_tx_hash(tx_message)
                 tx_message_response = TransactionMessage.respond_signing(
                     tx_message,
                     performative=TransactionMessage.Performative.SUCCESSFUL_SIGNING,
-                    tx_signature=tx_signature,
+                    signed_payload={"tx_signature": tx_signature},
                 )
-            else:
-                tx_message_response = TransactionMessage.respond_signing(
-                    tx_message,
-                    performative=TransactionMessage.Performative.REJECTED_SIGNING,
-                )
-            self.message_out_queue.put(tx_message_response)
-        else:
-            if self._is_acceptable_for_signing(tx_message):
-                tx_signature = self._sign_tx(tx_message)
+            if self._is_valid_tx(tx_message):
```

Comment:
We are checking the same thing in the line 755. I can see that we change the `tx_message` in the previous block in case it is true.

Reply:
Yes but for now I prefer it this way until I am refactoring the dm this week

Reply:
Ok. Makes sense :)